modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceShipmentItemLocalServiceImpl.java

```diff
@@ -83,14 +87,32 @@ public CommerceShipmentItem addCommerceShipmentItem(
 			0, serviceContext.getCompanyId(), externalReferenceCode);
 
 		CommerceOrderItem commerceOrderItem =
-			_commerceOrderItemLocalService.getCommerceOrderItem(
+			_commerceOrderItemLocalService.fetchCommerceOrderItem(
 				commerceOrderItemId);
 
+		if (commerceOrderItem == null) {
+			commerceOrderItem =
+				_commerceOrderItemLocalService.
+					fetchCommerceOrderItemByExternalReferenceCode(
+						externalReferenceCode, serviceContext.getCompanyId());
+		}
+
+		if (commerceOrderItem == null) {
+			throw new NoSuchOrderItemException(
+				StringBundler.concat(
+					"No commerce order item found with provided ID ",
+					commerceOrderItemId, " or external reference code ",
+					externalReferenceCode));
```

Comment:
where did you take this test? can you point me some examples? because is new for me and I would like to understand how is done in other places

Reply:
or, probably better so we can avoid this
1) go for the fetchCommerceOrderItemByExternalReferenceCode as first
2) if is null do the get by id (so it will raises the standard error)

Reply:
The problem is that let's say user does not provide an ID (it should work with ERC only too), then the user will get the ID error from the get which is incorrect because it was not provided in the first place. They should get the error that ERC is invalid.